NOMIS/nomis_test.go

```diff
@@ -0,0 +1,45 @@
+package main_test
+
+import (
+	nomis "github.com/ONSdigital/dp-dataset-api/NOMIS"
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+)
+
+func TestCheckSubString(t *testing.T) {
+	cases := []struct {
+		Description    string
+		GivenString    string
+		ExpectedResult string
+	}{
+		{
+			"Given a string with http and[Statistical Disclosure Control]",
+			"you can get the information from " +
+				"http://www.ons.gov.uk/statistical-disclosure-control/index.html[Statistical Disclosure Control] page on the ONS web site.",
+			"you can get the information from [Statistical Disclosure Control]" +
+				"(http://www.ons.gov.uk/statistical-disclosure-control/index.html) page on the ONS web site.",
+		},
+		{
+			"Given a string without [Statistical Disclosure Control]",
+			"you can get the information from " +
+				"http://www.ons.gov.uk/statistical-disclosure-control/index.aspx page on the ONS web site.",
+			"you can get the information from " +
+				"http://www.ons.gov.uk/statistical-disclosure-control/index.aspx page on the ONS web site.",
+		},
+		{
+			"Given a string without http and [Statistical Disclosure Control]",
+			"you can get the information from " +
+				"http://www.ons.gov.uk/statistical-disclosure-control/index.aspx page on the ONS web site.",
+			"you can get the information from " +
+				"http://www.ons.gov.uk/statistical-disclosure-control/index.aspx page on the ONS web site.",
+		},
+	}
+	Convey("Then the CheckSubString function should return the expected string", t, func() {
```

Comment:
The test is fine but I think the Convey statements are the wrong way around i.e. Then followed by Given. Minor tweak to this:
```
for _, test := range cases {
	Convey(test.Description, t, func() {
		Convey("Then the CheckSubString function should return the expected string", func() {
			actualString := nomis.CheckSubString(test.GivenString)
			So(actualString, ShouldResemble, test.ExpectedResult)
		})
	})
}
```